en/docs/learn/username-recovery-via-user-preferred-notification-channel.md

```diff
@@ -0,0 +1,353 @@
+
+
+!!! info "Notification Mechanisms"
+
+    WSO2 Identity Server provides the functionality to receive account recovery notifications 
+    internally or externally.
+    
+    1. **Internal Notification Management** : Notification sending is managed by WSO2 Identity Server.
+        
+    2. **External Notification Management** : Notification sending is managed by an external 
+    notification management mechanism.
+    
+    !!! note
+        To configure external notification management, `disable` the property `Enable Internal 
+        Notification Management` in `Account Recovery` configurations.
+        
+        ![disable-internal-notifications](../assets/img/learn/account-recovery/username-recovery/disable-internal-notification.png)
+    
+### Creating a user for recovery
+
+!!! tip "Before you begin" 
+    Make sure you have a user with Email or Mobile configured. If you already have 
+    a user, skip to the next heading. 
+    If not follow the steps below to create a new user and assign notification channels.
+
+1. Log in to the Management Console and click `Main -> Identity -> Users and Roles -> Add`.
+
+2. Click on `Add New User` and enter user credentials.
+
+    ![create-user](../assets/img/learn/account-recovery/add-user/create-user.png)
+    
+    !!! Note
+        For more details on creating users and roles Click [Adding Users and Roles](../.
+        ./learn/adding-users-and-roles/)
+    
+3. Navigate to `Main -> Identity -> Users and Roles -> List -> Users`.
+
+4. Find the user from the list and click `View Roles`.     
+    
+    ![user-entry](../assets/img/learn/account-recovery/add-user/navogate-to-default-profile.png)
+
+5. Click on `Permissions` to edit the default permissions.
+
+    ![get-list-of-permissions](../assets/img/learn/account-recovery/add-user/get-role-permissions.png)
+
+6. From the list of permissions select `Login` permission and click `Update`.     
+
+    ![add-login-permissions](../assets/img/learn/account-recovery/add-user/add-login-permissions.png) 
+    
+    !!! warning
+        This will update the permissions of the role. Therefore, all the users with the current role 
+        will receive login permissions.
+
+7. Navigate to `Main -> Identity -> Users and Roles -> List -> Users` and click on `User Profile` to 
+update the Email and Mobile of the user.
+
+    ![update-email-and-mobile](../assets/img/learn/account-recovery/add-user/add-email-and-mobile.png)
+    
+### **Username recovery with Internal Notification Management**   
+
+1. Use the following command to create a username recovery request.
+
+    ```
+    curl -X POST "https://localhost:9443/api/users/v1/recovery/username/init" -H "accept: application/json" -H "Content-Type: application/json" -d "{\"claims\":[{\"uri\":\"http://wso2.org/claims/givenname\",\"value\":\"user1\"}],\"properties\":[{\"key\":\"key\",\"value\":\"value\"}]}"
+    ```
+     
+2.  Following response will be returned by the API.
```

Comment:
Responses do not count as a step as it is not an action that the user needs to do. It can be given as part of the 1st step or the request and response can be given in code tabs